### ISSUE TITLE
msgp:shim ignored for time.Time

### ISSUE THREAD

**Issue by ebfe** [tinylib/msgp#94]
shims seem to be ignored for `time.Time`.

```go
package tt

import "time"

//go:generate msgp

//msgp:shim time.Time as:string using:timetostr/strtotime
type T struct {
	T time.Time
}

func timetostr(t time.Time) string {
	return t.Format(time.RFC3339)
}

func strtotime(s string) time.Time {
	t, _ := time.Parse(time.RFC3339, s)
	return t
}
```

The generated code still calls `dc.ReadTime` which expects a TimeExtension.

**Comment by philhofer** [comment]
I've had some time to look at this, and I've diagnosed the problem, but I haven't had time to put together a fix.

The shim code just sets the root reference with the given name in the file set to point to a different object. In practice, this doesn't work; the shim code would need to walk the entire set of identities and find all of them that had the right name and replace them.

**Comment by sintanial** [comment]
please, help me, how to solve this bug ? it's very important for me

**Comment by philhofer** [comment]
@sintanial I just put a fix out; see #142
